hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -7067,6 +7073,23 @@ public List<Fetch> visitFetches(FetchParent fetchParent) {
 		return fetches;
 	}
 
+	private boolean canFetchCreateALoop(Integer maxDepth, Fetchable fetchable) {
+		if ( maxDepth == null ) {
+			if ( fetchable instanceof ToOneAttributeMapping ) {
+				return this.isAssociationKeyVisited(
+						( (ToOneAttributeMapping) fetchable ).getForeignKeyDescriptor().getAssociationKey()
+				);
+			}
+			else if ( fetchable instanceof PluralAttributeMapping ) {
+				return this.isAssociationKeyVisited(
+						( (PluralAttributeMapping) fetchable ).getKeyDescriptor().getAssociationKey()
+				);
+			}
```

Comment:
Can you please add a comment here about why the `explicitFetch` should be set to false when "a fetch can create a loop"?

Reply:
we just do not change the `explicitFetch` value, when `explicitFetch` is true we do not do the circulatiry check

Reply:
so the method check if the fetch can create a look, in such a case we do not want to set `explicitFetch` to true in case it was not explicitly fetched though an explicit hql fetch